scripts/entities_extended_types_aliases/build_extended_aliases.ts

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env tsx
+import { writeFile } from 'node:fs/promises'
+import { difference, intersection, uniq, values, omit, cloneDeep } from 'lodash-es'
+import { prefixifyWd, getWdEntityUriNumericId } from '#controllers/entities/lib/prefix'
+import { makeCachedSparqlRequest } from '#data/wikidata/make_sparql_request'
+import { absolutePath } from '#lib/absolute_path'
+import { cache_ } from '#lib/cache'
+import { newError } from '#lib/error/error'
+import { oneDay } from '#lib/time'
+import { getHashCode, objectEntries } from '#lib/utils/base'
+import { info, logError } from '#lib/utils/logs'
+import { primaryTypesAliases, type PluralizedEntityType } from '#lib/wikidata/aliases'
+import { extendedAliasesQueries, getExtendedAliasesQueriesHash } from '#scripts/entities_extended_types_aliases/extended_type_aliases_queries'
+import type { WdEntityId, WdEntityUri } from '#server/types/entity'
+
+const extendedTypesAliases = cloneDeep(primaryTypesAliases)
+const stats = {}
+
+// Force a cache refresh by prefixing this script call with `export INV_REFRESH_ENTITIES_TYPE_EXTENDED_ALIASES=true;`
+const refresh = process.env.INV_REFRESH_ENTITIES_TYPE_EXTENDED_ALIASES === 'true'
+
+for (const [ type, sparql ] of objectEntries(extendedAliasesQueries)) {
+  let typeExtendedAliases = await getTypeExtendedAliases(type, sparql)
+  // Ideally, the following adjustments should be done within the queries
+  // but not at the cost of risking to time out
+  if (type === 'series') {
+    typeExtendedAliases = difference(typeExtendedAliases, extendedTypesAliases.collections)
+  } else if (type === 'works') {
+    typeExtendedAliases = difference(typeExtendedAliases, [ ...extendedTypesAliases.series, ...extendedTypesAliases.collections ])
+  }
+  if (type === 'movements' || type === 'genres') {
+    typeExtendedAliases = dropOverlaps(type, typeExtendedAliases)
+  } else {
+    checkOverlaps(type, typeExtendedAliases)
+  }
+  extendedTypesAliases[type] = typeExtendedAliases.sort(byNumericId)
+  stats[type] = typeExtendedAliases.length
+}
+
+info(stats, 'entity types aliases counts')
+
+async function getTypeExtendedAliases (type: PluralizedEntityType, sparqlRequests: string | string[]) {
+  const sparqlRequestsArray = sparqlRequests instanceof Array ? sparqlRequests : [ sparqlRequests ]
+  const hashCode = getHashCode(sparqlRequestsArray.join())
+  let extendedUris
+  try {
+    extendedUris = await cache_.get<WdEntityUri[]>({
+      // Use a a hash code in the key to bust outdated results when the query changes
+      key: `extended-wd-aliases:${type}:${hashCode}`,
+      fn: () => makeQueries(type, sparqlRequestsArray),
+      ttl: oneDay,
```

Comment:
Seems short, no ? Maybe this extended aliases is moving much more than I think, but `oneMonth` seems okay for a cache bust (doigt mouillé style)